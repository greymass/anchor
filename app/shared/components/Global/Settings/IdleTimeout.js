// @flow
import React, { Component } from 'react';
import { Dropdown } from 'semantic-ui-react';
import { translate } from 'react-i18next';

class GlobalSettingsIdleTimeout extends Component<Props> {
  onChange = (e, { value }) => {
    const { actions } = this.props;

    actions.setSetting('idleTimeout', value);
  }

  render() {
    const {
      defaultValue,
      name,
      selection,
      t
    } = this.props;

    const timeoutOptions = [
      { key: 'no_timeout', value: 999999999, text: t('global_settings_idle_timeout_no_timeout') },
      { key: 'five_minutes', value: 5 * 60 * 1000, text: t('global_settings_idle_timeout_five_minutes') },
      { key: 'fifteen_minutes', value: 15 * 60 * 1000, text: t('global_settings_idle_timeout_fifteen_minutes') },
      { key: 'thirty_minutes', value: 30 * 60 * 1000, text: t('global_settings_idle_timeout_thirty_minutes') },
      { key: 'sixty_minutes', value: 60 * 60 * 1000, text: t('global_settings_idle_timeout_sixty_minutes') }
    ];

    return (
      <Dropdown
        name={name}
        onChange={this.onChange}
        options={timeoutOptions}
        selection={selection}
        value={defaultValue || 999999999}
      />
    );
  }
}

export default translate('global')(GlobalSettingsIdleTimeout);
