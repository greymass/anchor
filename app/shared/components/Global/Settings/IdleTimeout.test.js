
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalSettingsIdleTimeout } from './IdleTimeout';

const mockProps = {
  defaultValue: 'default',
  name: 'transfer filter',
  selection: false,
  t: text => text,
};

describe('GlobalSettingsIdleTimeout', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalSettingsIdleTimeout {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
