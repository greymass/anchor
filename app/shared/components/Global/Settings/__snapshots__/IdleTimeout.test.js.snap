// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSettingsIdleTimeout renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  icon="dropdown"
  minCharacters={1}
  name="transfer filter"
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "key": "no_timeout",
        "text": "global_settings_idle_timeout_no_timeout",
        "value": 999999999,
      },
      Object {
        "key": "five_minutes",
        "text": "global_settings_idle_timeout_five_minutes",
        "value": 300000,
      },
      Object {
        "key": "fifteen_minutes",
        "text": "global_settings_idle_timeout_fifteen_minutes",
        "value": 900000,
      },
      Object {
        "key": "thirty_minutes",
        "text": "global_settings_idle_timeout_thirty_minutes",
        "value": 1800000,
      },
      Object {
        "key": "sixty_minutes",
        "text": "global_settings_idle_timeout_sixty_minutes",
        "value": 3600000,
      },
    ]
  }
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={false}
  value="default"
  wrapSelection={true}
/>
`;
